src/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -429,6 +429,95 @@ public static bool TrySkipBlock(BinaryReader reader)
 
             return true;
         }
+
+        public static bool TryValidateBlock(BinaryReader reader, ZipArchiveEntry entry)
+        {
+            const int OffsetToFilename = 26; // from the point after the signature
+
+            if (reader.ReadUInt32() != SignatureConstant)
+                return false;
```

Comment:
Are all 4 bytes guaranteed to be present?  If not, and `reader.ReadUInt32()` throws, is the exception that would throw consistent with what the calling code would already have thrown?

Reply:
I believe much of this code needs up-front checks for backing stream size to be consistent with what was done before.  See line 417, 425.

Reply:
yes @bartonjs it is guaranteed to have local file header signature, if not it is wrong zip format